### PR TITLE
Rewrite the settings in vue

### PR DESCRIPTION
**Pull request by artonge**
This is a rewrite of the settings in Vue.js

- [x] Switch to initial state
- [x] Convert templates to Vue
- [x] Create a store for the settings
- [x] Add Cypress tests
- [x] Delete legacy files

### User settings

Before | After
-- | --
Screenshot 2021-07-01 at 15-16-18 Settings - Nextcloud | Screenshot 2021-07-01 at 15-19-10 Settings - Nextcloud

### Admin settings

Before | After
-- | --
Screenshot 2021-07-01 at 15-16-29 Settings - Nextcloud | Screenshot 2021-07-01 at 15-19-20 Settings - Nextcloud

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -8,5 +8,6 @@ module.exports = {
 		'node/no-unpublished-import': ['error', {
 			allowModules: ['@vue/test-utils', '@testing-library/vue'],
 		}],
+		'valid-jsdoc': 'off',
```
Comment (artonge) [inline .eslintrc.js:11]: Needed until a new version of nextcloud/eslint-config is released

File: src/components/ActivitySettingsForm.vue [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<template>
+	<div>
+		<!-- TODO: should I use the global grid class ? -->
```
Comment (artonge) [inline src/components/ActivitySettingsForm.vue:24]: @skjnldsv ? An opinion ?

File: src/settings-store.js [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+		 * @param {Object} payload - The payload.
+		 * @param {boolean} payload.activityDigestEnabled - Enabled status of the activity digest.
+		 */
+		TOGGLE_ACTIVITY_DIGEST(state, { activityDigestEnabled }) {
+			state.activityDigestEnabled = activityDigestEnabled
+		},
+		/**
+		 * Toggle the availability of mail notifications
+		 *
+		 * @param {import('vuex').ActionContext<SettingsState, SettingsState>} store -
```
Comment (artonge) [inline src/settings-store.js:132]: Ugly way to add typing from a node_module

**Comment by artonge** [comment]
@skjnldsv the background around an hovered checkbox is not round, is it on purpose ?

Screenshot from 2021-06-30 19-14-37

File: src/models/NotificationsFrequency.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+export default {
+	NotificationsFrequency: {
```
Comment (nickvergessen) [inline src/models/NotificationsFrequency.js:24]: Please call it emails, activities also trigger notifications and it might be confusing than

**Review by nickvergessen — COMMENTED** [review]
Looks good but didnt test

File: src/models/NotificationsFrequency.js (hunk 4 above)
Comment (artonge) [reply src/models/NotificationsFrequency.js]: Done.

**Comment by AndyScherzinger** [comment]
@szaimen please don't add PRs or issues to the kanban board, this should only be done in and in alignment with @skjnldsv

**Comment by CarlSchwan** [comment]
This was integrated in another PR

**Pull request closed without merging**